packages/workflow/test/ExpressionExtensions/ObjectExtensions.test.ts

```diff
@@ -126,7 +127,9 @@ describe('Data Transformation Functions', () => {
 			test('should not allow prototype pollution', () => {
 				['{__proto__: {polluted: true}}', '{constructor: {prototype: {polluted: true}}}'].forEach(
 					(testExpression) => {
-						expect(evaluate(`={{ (${testExpression}).compact() }}`)).toEqual(null);
+						expect(() => evaluate(`={{ (${testExpression}).compact() }}`)).toThrow(
+							ApplicationError,
+						);
```

Comment:
The `tournament` hook `PrototypeSanitizer` throws on prototype pollution, whereas `tmpl` seems to skip the assignment. This slipped through the diff checker because it [suppresses](https://github.com/n8n-io/n8n/blob/313cfec74da58dc2403d828a2a211c6c4004f7df/packages/workflow/src/ExpressionEvaluatorProxy.ts#L80) any errors thrown, and because `ObjectExtensions` wasn't testing both evaluators.

Decided to adjust the assertion because it is unlikely users are relying a `null` coming from prototype pollution in an expression. There are no bug reports about this that I'm aware of. Please let me know if you disagree.